File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void Helper(List<Integer> ans, TreeNode root)
    {
        if(root == null)
            return;
        Helper(ans, root.left);
        Helper(ans, root.right);
        ans.add(root.val);
    }
    public int countNodes(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        Helper(ans, root);
        return ans.size();
    }
}

File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void Helper(vector<int>& ans, TreeNode* root)
    {
        if(root == NULL)
            return;
        Helper(ans, root->left);
        Helper(ans, root->right);
        ans.push_back(root->val);
    }
    int countNodes(TreeNode* root) {
        vector<int> ans;
        Helper(ans, root);
        return ans.size();
    }
};